api/discord_bot.go

```diff
@@ -0,0 +1,52 @@
+package api
+
+import (
+	"fmt"
+	"github.com/bwmarrin/discordgo"
+	"github.com/gin-gonic/gin"
+	"strings"
+)
+
+type DiscordBot struct {
+	Session *discordgo.Session
+}
+
+func CreateDiscordBot() *DiscordBot {
+	PanicIfDiscordBotMissesInformation()
+	session, _ := discordgo.New(fmt.Sprintf("Bot %s", DiscordToken))
+
+	return &DiscordBot{
+		Session: session,
+	}
+}
+
+func (bot *DiscordBot) SendMessage(context *gin.Context) {
```

Comment:
p-ê que cette méthode devrait retourner une erreur

Reply:
Dans ce cas-ci, on ne peut rien retourner. Le type HandlerFunc de gin a cette signature: func(*Context). Si on retourne une erreur, on ne pourra plus utiliser cette méthode comme route handler. On peut utiliser une approche alternative avec un context.AbortWithError ou context.AbortWithStatusJSON et après utiliser un middleware dans notre serveur pour dealer avec ces erreurs.

Reply:
Un middleware comme: https://github.com/gin-gonic/gin/issues/1952#issuecomment-508043499 serait vraiment sharp in this case